### ISSUE TITLE
Task 4 - Filter Out Certain Methods

### ISSUE THREAD

**Issue by sjiang1** [grandfoosier/EMUresearch#4]
Now that we have the python methods in place, I'd like to filter out some methods. In the paths.txt, you have all the methods' names.

Write a script to create a new paths_remove_test.txt with the same format as paths.txt, but exclude all the test methods. (later, we are going to exclude other types of methods.)

The basic filtering approach is 
> functions whose name contains the substring test
If you have a better idea, just let me know and we can decide what kind of filtering mechanism that we are going to adopt.

Thanks!

**Comment by grandfoosier** [comment]
I added a python file to filter the paths list, and pushed both the old and new lists.
I also changed it so in the future the program wouldn't save methods which don't meet the criteria.
I only did 'test' right now, but it should be easy to add other things to the filter.

**Comment by sjiang1** [comment]
Could you save the excluded methods (that don't meet the criteria) in a new file, like paths_filtered.txt, even the excluded methods are not saved locally? For each criterion, we need a new file. Thanks!

**Comment by grandfoosier** [comment]
Just to clarify, should I leave in the code to not save the excluded methods? Just move the paths from the full list to different lists?

**Comment by sjiang1** [comment]
That's correct. You can leave the code to not save the excluded methods. Move the paths from the full list to different lists OR have a flag in the full list to indicate whether and why a method is excluded. Either way is fine with me, and you can decide which way is easier for processing.

**Comment by sjiang1** [comment]
Looking at the paths_filtered.txt, I just realized some test files don't have 'test' in their filenames, but they have parent directories named with 'test'. Would you update your script to consider this situation?

Thanks!
